tests/run-macros/beta-reduce-inline-result/Test_2.scala

```diff
@@ -0,0 +1,12 @@
+import scala.compiletime._
+
+object Test {
+  
+  inline def dummy: Int => Int =
+    (i: Int) => i + 1
+
+  def main(argv : Array[String]) : Unit = {
+    println(code"${Macros.betaReduce(dummy)(3)}")
+  }
+}
+
```

Comment:
We could have a simple neg test `tests/neg-macros/beta-reduce-inline-result/Test_2.scala` (using a copy of `beta-reduce-inline-result/Macro_1.scala`) to check that the elaboration is not changed.

```scala
import scala.compiletime._

object Test {

  inline def dummy: Int => Int =
    (i: Int) => i + 1
  
  val x1: Int = Macros.betaReduce(dummy)(3)
  val x2: 4   = Macros.betaReduce(dummy)(3) // error
}
```

Reply:
Agreed, I will add this.

Reply:
Funny detail: this test actually passes already because the inline expansion adds its own type ascription. I'll look for a test that properly exercises the case where betaReduce's own type ascription is needed...

Reply:
... actually @LPTK 's test should handle this. Thanks @LPTK :+1: